lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -132,11 +134,16 @@ export abstract class ClientApplication {
      */
     async acquireTokenByCode(request: AuthorizationCodeRequest, authCodePayLoad?: AuthorizationCodePayload): Promise<AuthenticationResult | null> {
         this.logger.info("acquireTokenByCode called", request.correlationId);
+        if (request.state && authCodePayLoad){
+            this.validateState(request.state, authCodePayLoad.state || "");
+            authCodePayLoad.state = undefined;
```

Comment:
We should not mutate variables, why is this needed?

Reply:
When state is added to authCodePaylLoad, msal common's ProtocolUtils' parseRequestState method parses the library state and state sent by the user. In msal node we are not using library state, so to avoid ClientAuthError.createInvalidStateError, I am setting the state to undefined.

Reply:
If you need to do this you should create a new variable (using the spread operator) rather than editing the one passed in